e2e/testcases/multi_sync_test.go

```diff
@@ -574,14 +577,34 @@ func TestConflictingDefinitions_RootToRoot(t *testing.T) {
 	nt.T.Logf("Stop the admission webhook, to confirm that the remediator still reports the conflicts")
 	nomostest.StopWebhook(nt)
 
-	nt.T.Logf("The Role resource version should be changed because two reconcilers are fighting with each other")
-	err = nt.Watcher.WatchObject(kinds.Role(), "pods", testNs,
-		[]testpredicates.Predicate{
-			testpredicates.ResourceVersionNotEquals(nt.Scheme, roleResourceVersion),
-		})
-	if err != nil {
-		nt.T.Fatal(err)
-	}
+	// Watch Role until the other conditions are met.
+	// Collect data from role updates for validation later
+	// This will log diffs for easier debugging.
+	var roleUpdates []roleUpdate
+	nt.T.Log("Starting Role Watcher")
+	bgCtx, bgCancel := context.WithCancel(nt.Context)
+	defer bgCancel()
+	bgDone := make(chan struct{})
+	go func() {
+		defer close(bgDone)
+		err = nt.Watcher.WatchObject(kinds.Role(), "pods", testNs,
+			[]testpredicates.Predicate{
+				func(o client.Object) error {
+					if o != nil {
+						ru := roleUpdate{
+							ResourceVersion: o.GetResourceVersion(),
+							Manager:         core.GetAnnotation(o, metadata.ResourceManagerKey),
+						}
+						roleUpdates = append(roleUpdates, ru)
+						nt.T.Logf("Role update: %#v", ru)
+					}
+					return fmt.Errorf("waiting for other conditions")
+				},
+			}, testwatcher.WatchTimeout(0), testwatcher.WatchContext(bgCtx))
+		if err != nil && !errors.Is(err, context.Canceled) {
```

Comment:
nit: err should always be != nil. I think it would be simpler/clearer to just write

```
if !errors.Is(err, context.Canceled) {
```

Reply:
obsolete.